src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/ProjectSystem/RemoteProjectSnapshot.cs

```diff
@@ -218,11 +214,18 @@ private async Task<RazorConfiguration> CreateRazorConfigurationAsync()
 
         var suppressAddComponentParameter = compilation is not null && !compilation.HasAddComponentParameter();
 
+        var csharpParseOptions = _project.ParseOptions as CSharpParseOptions ?? CSharpParseOptions.Default;
+        var csharpLanguageVersion = csharpParseOptions.LanguageVersion;
+        var useRoslynTokenizer = csharpParseOptions.UseRoslynTokenizer();
+
         return new(
             razorLanguageVersion,
             configurationName,
             Extensions: [],
             UseConsolidatedMvcViews: true,
-            suppressAddComponentParameter);
+            suppressAddComponentParameter,
+            CSharpLanguageVersion: csharpLanguageVersion,
+            UseRoslynTokenizer: useRoslynTokenizer,
+            RootNamespace: _project.DefaultNamespace);
```

Comment:
Should this include `?? "ASP"` like it did before?

```suggestion
            RootNamespace: _project.DefaultNamespace ?? "ASP");
```

Reply:
I wonder if `RazorConfiguration.RootNamespace` should be non-null, and default to "ASP".

Reply:
That seems a reasonable suggestion to me, but would that require more changes in the compiler to be sure nothing breaks? Without doing a deeper survey of the compiler, I don't know if there isn't something that allows `RootNamespace` to be null. For example, this change adds `RazorConfiguration.RootNamespace` but nothing in the compiler actually uses that yet. Instead, everything still passes that value into the `SetRootNamespace` extension method. If it defaults to `"ASP"` and we change the compiler to just read `RazorConfiguration.RootNamespace`, would that break something that _didn't_ previously call `SetRootNamespace` during configuration?

Reply:
You make a very convincing argument.

Reply:
The argument can be summarized, "I'll support you from afar if you break into jail!" 🤣